src/main/scala/swiss/dasch/api/BaseEndpoints.scala

```diff
@@ -5,34 +5,45 @@
 
 package swiss.dasch.api
 
+import cats.implicits._
 import sttp.model.StatusCode
 import sttp.model.headers.WWWAuthenticateChallenge
+import sttp.tapir.Codec
+import sttp.tapir.EndpointOutput
+import sttp.tapir.PublicEndpoint
 import sttp.tapir.generic.auto.*
 import sttp.tapir.json.zio.jsonBody
+import sttp.tapir.oneOf
+import sttp.tapir.oneOfVariant
+import sttp.tapir.statusCode
 import sttp.tapir.ztapir.*
-import sttp.tapir.{Codec, EndpointOutput, PublicEndpoint, oneOf, oneOfVariant, statusCode}
+import swiss.dasch.api.ApiProblem.Unauthorized
 import swiss.dasch.api.BaseEndpoints.defaultErrorOutputs
+import swiss.dasch.domain.AuthScope
+import zio.IO
 import zio.ZLayer
 
-case class UserSession(subject: String)
+case class UserSession(
+  subject: String,
+  scope: AuthScope,
+)
 
 case class BaseEndpoints(authService: AuthService) {
-
   val publicEndpoint: PublicEndpoint[Unit, ApiProblem, Unit, Any] = endpoint
     .errorOut(defaultErrorOutputs)
 
-  val secureEndpoint: ZPartialServerEndpoint[Any, String, UserSession, Unit, ApiProblem, Unit, Any] = endpoint
+  val jwtAuthenticatedEndpoint: ZPartialServerEndpoint[Any, String, UserSession, Unit, ApiProblem, Unit, Any] = endpoint
     .errorOut(defaultErrorOutputs)
```

Comment:
**nitpick**: The old name was fine and did not expose implementation details

For specifying an endpoint in the dsp-ingest we currently have only two different options.
1. An unprotected _public_ endpoint
2. A protected _secure_ endpoint which needs authentication and possibly authorization

I do not think we need to include the means of authorization (JWT) in the to the name of the secure endpoint. Every endpoint handler will anyway only receive the principal, i.e. `Usersession` for further authorization.

Reply:
✔️